src/requestPool/requestPoolManager.js

```diff
@@ -175,21 +221,26 @@ function startGrabbing () {
 }
 
 function getNextRequest () {
-  if (requestPool.interaction.length && numRequests.interaction < maxNumRequests.interaction) {
-    return requestPool.interaction.shift();
-  }
+  let stillRequest = false;
+
+  for (let i = 0; i < requestPoolTypes.length; i++) {
+    const name = requestPoolTypes[i].name;
 
-  if (requestPool.thumbnail.length && numRequests.thumbnail < maxNumRequests.thumbnail) {
-    return requestPool.thumbnail.shift();
+    if (requestPool[name].length && numRequests[name] < getMaxRequests(i)) {
+      return requestPool[name].shift();
+    }
   }
 
-  if (requestPool.prefetch.length && numRequests.prefetch < maxNumRequests.prefetch) {
-    return requestPool.prefetch.shift();
+  for (let i = 0; i < requestPoolTypes.length; i++) {
```

Comment:
I wonder. Do we _need_ to iterate over each request type again? In the first loop, we already check for `requestPool[name].length`. We could change...

```
if (requestPool[name].length && numRequests[name] < getMaxRequests(i)) {
    return requestPool.thumbnail.shift();
}
```

To something like...

```
let hasRequestsInQueue = false;

const hasPooledRequests = requestPool[name].length > 0;
const isUnderMaxActiveRequests = ...
if(hasPooledRequests && isUnderMaxActiveRequests){
   ...
}else if(hasPooledRequests) {
    hasRequestsInQueue = true;
}
```

Or at least something like: [Array.Prototype.Every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)

Reply:
Done :)